llvm/test/tools/sycl-post-link/virtual-functions/dummy.ll

```diff
@@ -0,0 +1,37 @@
+; RUN: sycl-post-link -split=auto -properties -S < %s -o %t.table
+; RUN: FileCheck %s --input-file=%t.table --check-prefix=CHECK-TABLE
+; RUN: FileCheck %s --input-file=%t_0.ll --check-prefix=CHECK-FP64-SPLIT
+; RUN: FileCheck %s --input-file=%t_1.ll --check-prefix=CHECK-FP64-DUMMY
+; RUN: FileCheck %s --input-file=%t_1.prop --check-prefix=CHECK-FP64-DUMMY-PROPS
+; RUN: FileCheck %s --input-file=%t_2.ll --check-prefix=CHECK-FP32-SPLIT
+
+; CHECK-TABLE:      _0.prop
+; CHECK-TABLE-NEXT: _1.prop
+; CHECK-TABLE-NEXT: _2.prop
+
+; CHECK-FP64-SPLIT: define spir_func void @bar()
+; CHECK-FP32-SPLIT: define spir_func void @foo()
+
+; CHECK-FP64-DUMMY: define spir_func void @bar()
+; CHECK-FP64-DUMMY-NEXT: entry:
+; CHECK-FP64-DUMMY-NEXT: ret void
+
+; CHECK-FP64-DUMMY-PROPS: dummy=1
```

Comment:
I would also check that we have not emitted the property for other device images. And I would also throw a virtual function without any optional kernel features on it into the mix as well